ocs_ci/helpers/dr_helpers.py

```diff
@@ -562,213 +561,264 @@ def check_mirroring_status_for_custom_pool(
-    rd_obj = ocp.OCP(kind=constants.REPLICATIONDESTINATION, namespace=namespace)
-    rd_items = rd_obj.get().get("items")
-    return len(rd_items)
+    resource_obj = ocp.OCP(kind=kind, namespace=namespace)
+
+    try:
+        if resource_name:
+            return resource_obj.is_exist(resource_name=resource_name)
+        else:
+            resource_items = resource_obj.get().get("items", [])
+            return len(resource_items) > 0
+    except Exception as e:
+        if "Error from server (NotFound)" in str(e):
+            logger.info(f"{kind} {resource_name} not found in namespace {namespace}.")
+        else:
+            logger.warning(f"Exception raised when fetching {kind}: {e}")
+        return False
 
 
-def check_vr_state(state, namespace):
+def check_replication_resource_state(kind, state, namespace, resource_name=""):
     """
-    Check if all VRs in the given namespace are in expected state
+    Check if replication resources in the given namespace are in expected state
 
     Args:
-        state (str): The VRs state to check for (e.g. 'primary', 'secondary')
-        namespace (str): the namespace of the VR resources
+        kind (str): Kind of resource (e.g., constants.VOLUME_REPLICATION, constants.VOLUME_REPLICATION_GROUP, etc.)
+        state (str): The resource state to check for (e.g. 'primary', 'secondary')
+        namespace (str): the namespace of the resources
+        resource_name (str): Name of specific resource
 
     Returns:
-        bool: True if all VRs are in expected state or were deleted, False otherwise
+        bool: True if resources are in expected state or were deleted, False otherwise
 
     """
-    vr_obj = ocp.OCP(kind=constants.VOLUME_REPLICATION, namespace=namespace)
-    vr_list = vr_obj.get().get("items")
+    resource_obj = ocp.OCP(kind=kind, namespace=namespace)
 
-    # Skip state check if resource was deleted
-    if len(vr_list) == 0 and state.lower() == "secondary":
-        logger.info("VR resources not found, skipping state check")
-        return True
+    if resource_name:
+        resource_list = resource_obj.get(resource_name=resource_name)
+        resource_list_index = resource_list
+    else:
+        resource_list = resource_obj.get().get("items")
+        resource_list_index = resource_list[0] if resource_list else None
+
+    # Handle deletion case
+    if len(resource_list) == 0 and state.lower() == "secondary":
+        if kind == constants.VOLUME_REPLICATION_GROUP:
+            ocs_version = version.get_semantic_ocs_version_from_config()
+            if ocs_version <= version.VERSION_4_17:
+                logger.info(f"{kind} resource not found, skipping state check")
+                return True
+            else:
+                logger.info(f"{kind} resource not found")
+                return False
+        else:
+            logger.info(f"{kind} resource not found, skipping state check")
+            return True
+
+    if kind == constants.VOLUME_REPLICATION:
```

Comment:
We need to include another condition like len(resource_list) > 1, to execute "# Handle multiple VR resources" only.
In the current code, it looks like line no 674 will be executed even when there is a single VR